tool/site/client/src/services/test-packs/xls.ts

```diff
@@ -2,7 +2,7 @@ import {Dispatch, SetStateAction} from "react"
 import * as XLSX from "xlsx"
 
 export interface XlsRow {
-  [column: string]: string
+  [column: string]: string | undefined
```

Comment:
You can't use optional types with generic key values e.g. `[column: string]`

This is because implicitly any particular key in the `XlsRow` type is optional, for example:

`
interface Row {
  key1?: string
  key2?: string
}

const row1: Row = {
  key1: "key1 value"
}

const row2: Row = {
  key2: "key2 value"
}
`

Reply:
ah makes sense, would it be more appropriate to type it as `string | null` then?

Reply:
feel like `string | null` is more consistent with the code we have atm

Reply:
The value that is returned by xls when reading a empty cell from a spreadsheet is `undefined` rather than `null`. For all practical reasons I think `null` would be fine but I'd prefer the types to be as accurate as possible